common/lib/symbiosis/domain_hooks.rb

```diff
@@ -0,0 +1,45 @@
+require 'symbiosis'
+require 'open3'
+
+module Symbiosis
+  # generic hooks implementation
+  # use subclasses to provide default behaviour
+  # see Symbiosis::DomainSkeleton::Hooks
+  # and Symbiosis::SSL::Hooks
+  class DomainHooks
```

Comment:
This could be just a generic Symbiosis::Hooks class with `run!` arguments changed to be just an array of strings, as discussed :)